### ISSUE TITLE
As a Christmas tree applicant I would like the opportunity to verify or make sure I have an version of the permit via email or direct download after payment.

### ISSUE THREAD

**Issue by ASprinkle** [18F/fs-open-forest-platform#489]
## Notes
The team has received a few issues around receiving a permit via email.  The issue has been the result of the user entering their email address incorrectly.

## Acceptance Criteria
- [x] Applicant has an opportunity to verify their email address
- [] Applicant can download their permit when they choose to print it from Open Forest.

## Tasks
- [x] Design (effective, not intrusive).
- [] Integration pm Research for clicking to download permit at Open Forest confirmation page.
- [] Implementing the download feature if research suggests.
- [x]  Compile email failure messages for context.
- [x]  Build
- [x] PO approved.

## Definition of Done
- [x] Code unit / e2e tested and comply with QWASP
- [x] Server actions captured by logs
- [x] No new security vulnerabilities
- [x] Internal team code review
- [x] Documentation / readme.md / CircleCI updated
- [x] Docker updated
- [x] Compare finished design with mockup
- [x] Usability tested

**Comment by lisaredux** [comment]
2nd recommendation - Permit details page:

Recommend adding another input to put email address twice as a confirmation. Extra validation to ensure the email addresses match would have to be there in addition to extra validation of the form to go through. 

<img width="342" alt="screen shot 2018-12-05 at 5 44 45 pm" src="https://user-images.githubusercontent.com/26801590/49549060-a40ec300-f8b5-11e8-8782-2a690bef2153.png">

**Comment by ASprinkle** [comment]
If possible it might be good to do a mix of both - I like the set up for the 2nd recommendation. For the first one, i might rephrase the note to say "If you've entered your email address in correctly, print your permit with the rules and guidelines here." or something like that.

**Comment by ASprinkle** [comment]
@lisaredux  per this morning's stand up we will proceed with the language to "Please verify your email address" box on the permit holder information page.  The team will implement the option to download direct only if issues continue to present.  

What is the next step to move the changes into PO viewable?

**Comment by ASprinkle** [comment]
Lisa and Mark will provide design as follow up from our meeting yesterday.

**Comment by lisaredux** [comment]
> @lisaredux per this morning's stand up we will proceed with the language to "Please verify your email address" box on the permit holder information page. The team will implement the option to download direct only if issues continue to present.
> 
> What is the next step to move the changes into PO viewable?

<img width="262" alt="screen shot 2018-12-14 at 10 40 56 am" src="https://user-images.githubusercontent.com/26801590/50012506-d80c7700-ff8c-11e8-9e39-3bf4049f1672.png">

Update made. This treatment assumes that the first email box is getting validated as a match against the second.

**Comment by ASprinkle** [comment]
From stand up 12/17:  the next step is to get this design into staging - will need input and work from dev.

**Comment by ASprinkle** [comment]
I updated the AC and the tasks based on the ^ discussion threads.

**Comment by lisaredux** [comment]
For implementing on the confirmation page: REORDERING CONTENT
<img width="471" alt="screen shot 2019-02-22 at 10 48 46 am" src="https://user-images.githubusercontent.com/26801590/53253583-99332580-368f-11e9-875f-82a5c3fe4396.png">
or
<img width="471" alt="screen shot 2019-02-22 at 10 48 53 am" src="https://user-images.githubusercontent.com/26801590/53253592-9c2e1600-368f-11e9-8c56-03f5fa2316df.png">

**Comment by lisaredux** [comment]
For implementing on the permit details page: REORDERING CONTENT + ADDING VERIFICATION EMAIL INPUT FIELD TO BE VALIDATED THAT THEY MATCH
<img width="384" alt="screen shot 2019-02-22 at 11 30 00 am" src="https://user-images.githubusercontent.com/26801590/53256394-c2ef4b00-3695-11e9-8518-211188fdb61c.png">

**Comment by davemcorwin** [comment]
@lisaredux Do you want the email confirmation to also appear on the Special Uses forms?

**Comment by ASprinkle** [comment]
Verification is successful when entering email address in and skipping verification box.
Verification is successful when entering email address into the verification box and address differs from the first box.

**Comment by lisaredux** [comment]
Working on usability test plan

**Comment by lisaredux** [comment]
> @lisaredux Do you want the email confirmation to also appear on the Special Uses forms?

@davemcorwin Just saw your question. Seems like a good idea to implement it — unless the email address is already getting captured and carried over from authentication step. I'll delve into it and have a better clue by Friday.

**Comment by lisaredux** [comment]
@davemcorwin @ASprinkle The first part of this story has been implemented with the email validation. Using credit card `4111111111111111` I was able to get to the confirmation screen. There are content updates need to be implemented to the confirmation page. 


What I see today:
<img width="781" alt="Screenshot 2019-04-19 10 31 17" src="https://user-images.githubusercontent.com/26801590/56428907-0aa5e380-628f-11e9-8685-befc2615d8c6.png">
<img width="795" alt="Screenshot 2019-04-19 10 31 28" src="https://user-images.githubusercontent.com/26801590/56428911-0d083d80-628f-11e9-8cb5-d31e40f74803.png">


What it should be: ![image](https://user-images.githubusercontent.com/26801590/56428076-1ba12580-628c-11e9-9aee-1c630df4428b.png)


Also, to answer your earlier question @davemcorwin, we definitely would like to implement the same email validation logic to Special Uses. Thanks for catching that. However, as it is now, there's an issue with the validation implementation. Basically, this validation comes too early before the user enters anything in the second email field <img width="541" alt="Screenshot 2019-04-19 09 57 55" src="https://user-images.githubusercontent.com/26801590/56428533-b6e6ca80-628d-11e9-8dbb-8db84fa0454c.png">

@ASprinkle @davemcorwin This validation issue is not a dealbreaker for this story, but will need to get fixed eventually and before implementing into Special Uses. Not sure how we should handle this — as a separate bug or in this story? Also, should we create a issue/story to implement email validation in Special Uses? With similar logic to Christmas trees?

**Comment by lisaredux** [comment]
Usability test plan for this story lives here: https://github.com/18F/fs-open-forest-platform/wiki/Usability-test-plan-for-%23489

**Comment by ASprinkle** [comment]
Story #600 is currently in the Backlog regarding entering invalid email addresses I have updated a task in that story to reflect the timing of the error note.

**Comment by ASprinkle** [comment]
For now, it is ok that the validation notification populates early.  The text displays the correct need as the field becomes filled out.  If this becomes an issue later, we can address the timing of the validation note for the email address data - created a new issue to capture the work.

**Comment by lisaredux** [comment]
In progress: https://app.mural.co/t/gsa6/m/gsa6/1556304246045/9078126a0403e523b60577ef1a3da304df563a83

**Comment by lisaredux** [comment]
Here's the updated text block that should go on the confirmation page. We can edit and dev can grab from here:

> 
> Your permit and the Christmas tree cutting rules and guidelines were emailed to you at <b>**email address**</b>. If for some reason this email address is incorrect, please ensure you print your permit below with rules and guidelines. 
> 
> You must print this permit and have it properly displayed on your vehicle when cutting down your Christmas tree. When printing your permit, use plain white paper. The printed permit will have instructions on how to properly place it on your vehicle. 
> 
> [Checkbox] Include a copy of the Christmas tree rules and guidelines
> 
> [Print button]
> 
![image](https://user-images.githubusercontent.com/26801590/57232886-99cf2d00-6feb-11e9-928d-406aec92fac3.png)

**Comment by ASprinkle** [comment]
When I tested this on staging for the ARP, it does not look like the content on the confirmation page is loading correctly for me. 
I did not test it for other forests.
It appears the new text in this story is missing, as well as the order of the headers and info being presented. Here is a screen shot of what I encountered:

![image](https://user-images.githubusercontent.com/33435621/59052357-c7560300-8843-11e9-9ea9-c20872cc2f2d.png)

**Comment by mtlaney** [comment]
Sorry I may have moved it over to soon. I had just merged it and it takes a few to build and deploy.

**Comment by ASprinkle** [comment]
@mtlaney  The content looks really good, 
One fix needed before approving regarding the order of the headers/info.
1st should be the Print Permit
2nd should be Order Summary
3rd should be Order Details
followed by Additional Information.

Here is what I saw on staging: 

![image](https://user-images.githubusercontent.com/33435621/59128965-55072080-8920-11e9-97db-79d7e96aafc2.png)


Thanks!

**Comment by ASprinkle** [comment]
@mtlaney The content looks good, let's rearrange so that: 
Print Permit (also add the line under the header) is first, followed by
the Order Summary, then
the Order Details, followed by
Additional Information.
[Story489screenshots.pdf](https://github.com/18F/fs-open-forest-platform/files/3272673/Story489screenshots.pdf)

**Comment by ASprinkle** [comment]
Hi @mtlaney not sure what the issue is, but there is still no change when I check it out in staging.
I also tried it in an incognito window, and had the same result.

![image](https://user-images.githubusercontent.com/33435621/59513110-07823a80-8e6f-11e9-9997-3219f4c2e0f4.png)

**Comment by davemcorwin** [comment]
@ASprinkle the deployment is pending a successful snyk run which is held up by https://code.fs.usda.gov/NFS/Open-Forest-Program/issues/64.

**Comment by mtlaney** [comment]
@ASprinkle Should be live and ready to test now on staging :)

**Comment by ASprinkle** [comment]
Success, looks good!
